synapse/lib/time.py

```diff
@@ -0,0 +1,44 @@
+'''
+Time related utilities for synapse "epoch millis" time values.
+'''
+
+import datetime
+
+from synapse.exc import BadTypeValu
+
+def parse(text, base=None):
```

Comment:
This is also very duplicative of the TimeType's norm str function.  Consider changing timetype to use this function, so we're not implementing the same logic twice.